Explain the customCheProperties field (#838)
codenvy/che-docs: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc

To edit the `che` `configMap`, edit the Custom Resource manually.
The `configMap` derives values from the `CheCluster` field. User modifications of the `CheCluster` Custom Resource field cause the Operator to change the attributes of the `che` `configMap` accordingly. The `configMap` changes automatically trigger a restart of the Che Pod.

To add custom properties to the Che server, such as environment variables that are not automatically generated in the `che` `configMap` by the Operator, or to override automatically generated properties, the `CheCluster` Custom Resource has a `customCheProperties` field, which expects a map.

For example, to add a property named `FOO` with value `BAR`, add an entry in `customCheProperties`:

[source,yaml]
----
apiVersion: org.eclipse.che/v1
kind: CheCluster
metadata:
  name: eclipse-che
  namespace: che
spec:
  server:
    cheImageTag: ''
    devfileRegistryImage: ''
    pluginRegistryImage: ''
    tlsSupport: false
    selfSignedCert: false
    customCheProperties:
      FOO: BAR  
  database:

>     externalDb: false

Reviewer: I'm not sure we need to add all the other fields in the other sections...
Maybe ... would be sufficient no ?

link:https://docs.openshift.com/container-platform/latest/applications/operators/olm-what-operators-are.html[Operators] are software extensions to Kubernetes that use link:https://docs.openshift.com/container-platform/latest/applications/crds/crd-managing-resources-from-crds.html[custom resources] to manage applications and their components.

> Che installed using the Operator provides the user with an automatically generated `che` `configMap`.

Reviewer 2: ```suggestion
Che installed using the Operator provides the user with an automatically generated `configMap` called `che`.
```

> The `che` `configMap` contains the main properties for the installation and is in sync with the information stored in the CheCluster Custom Resource file. User modifications of the `che` `configMap` after installing Che using the Operator are automatically overwritten by values that the Operator obtains from the `CheCluster` Custom Resource. 

Reviewer 2: ```suggestion
The `che` `configMap` contains the main properties for the Che server and is in sync with the information stored in the CheCluster Custom Resource file. User modifications of the `che` `configMap` after installing Che using the Operator are automatically overwritten by values that the Operator obtains from the `CheCluster` Custom Resource. 
```

* `che` (main)
* `custom` (additional)
The `che` `configMap` contains the main properties for the installation and is in sync with the information stored in the CheCluster Custom Resource file. User modifications of the `che` `configMap` after installing Che using the Operator are automatically overwritten by values that the Operator obtains from the `CheCluster` Custom Resource. 

> * The `che` `configMap` contains the main properties for the installation and is in sync with the information stored in the CheCluster Custom Resource file. User modifications of the `che` `configMap` after installing Che using the Operator are automatically overwritten by values that the Operator obtains from the `CheCluster` Custom Resource. +

Reviewer: I would replace:

*main properties __for the installation__*

by 

*main properties __for the Che server__*

> To add custom properties, such as environment variables that are not created by the `che` `configMap` by default, or to override standard properties, the `CheCluster` Custom Resource has a `overrideCheProperties` field, that expects a list of key/value pairs.  

Reviewer: ```suggestion
To add custom properties to the Che server, such as environment variables that are not automatically generated in the `che` `configMap` by the operator, or to override automatically generated properties, the `CheCluster` Custom Resource has a `customCheProperties` field, that expects a map.  
```

Reviewer 2: Building on David's suggestion:

```suggestion
To add custom properties to the Che server, such as environment variables that are not automatically generated in the `che` `configMap` by the Operator, or to override automatically generated properties, the `CheCluster` Custom Resource has a `customCheProperties` field, which expects a map.  
```

> For example, to add a property named `FOO` with value `BAR`, add an entry in `overrideCheProperties`:

Reviewer: ```suggestion
For example, to add a property named `FOO` with value `BAR`, add an entry in `customCheProperties`:
```

[source,yaml]
----
apiVersion: org.eclipse.che/v1
kind: CheCluster
metadata:
  name: eclipse-che
  namespace: che
spec:
  server:
    cheImageTag: ''
    devfileRegistryImage: ''
    pluginRegistryImage: ''
    tlsSupport: false
    selfSignedCert: false
    overrideCheProperties:
       - name: FOO
         value: BAR
  database:
    externalDb: false
    chePostgresHostname: ''
    chePostgresPort: ''
    chePostgresUser: ''
    chePostgresPassword: ''
    chePostgresDb: ''
  auth:
    openShiftoAuth: true
    identityProviderImage: ''
    externalIdentityProvider: false
    identityProviderURL: ''
    identityProviderRealm: ''
    identityProviderClientId: ''
  storage:
    pvcStrategy: per-workspace
    pvcClaimSize: 1Gi
    preCreateSubPaths: true
----

> Previous versions of the Che operator had a configMap named `custom` to fulfill this role.  If the Che operator finds a `configMap` with the name `custom`, it will add the data it contains to the `overrideCheProperties` field, redeploy Che, and delete the `configMap`. 

Reviewer: ```suggestion
Previous versions of the Che operator had a configMap named `custom` to fulfill this role.  If the Che operator finds a `configMap` with the name `custom`, it will add the data it contains into the `customCheProperties` field, redeploy Che, and delete the `custom` `configMap`. 
```

Reviewer 2: Building on David's suggestion:

```suggestion
Previous versions of the Che Operator had a configMap named `custom` to fulfill this role.  If the Che Operator finds a `configMap` with the name `custom`, it adds the data it contains into the `customCheProperties` field, redeploys Che, and deletes the `custom` `configMap`. 
```
